app/forms/upload_form.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+CSV_CONTENT_TYPE = 'text/csv'
+CSV_COUNT_MIN = 1
+CSV_COUNT_MAX = 1000
+
+class UploadForm
+  include ActiveModel::Validations
+
+  attr_reader :search_engine, :csv_file
+
+  validates :search_engine, presence: true
+  validates :search_engine, inclusion: { in: SearchResult.search_engines.keys }
+  validates :csv_file, presence: true
+
+  validate :valid_content_type, if: -> { csv_file.present? }
+  validate :valid_count, if: -> { csv_file.present? }
+
+  def initialize(search_engine:, csv_file:)
+    @search_engine = search_engine
+    @csv_file = csv_file
+  end
+
+  def valid_content_type
+    return if csv_file.content_type == CSV_CONTENT_TYPE
+
+    add_error :invalid_extension
+  end
+
+  def valid_count
+    return if CSV.read(csv_file.tempfile).count.between?(CSV_COUNT_MIN, CSV_COUNT_MAX)
```

Comment:
This might raise an error when you pass a non-CSV file. We should validate this when the `csv_file` is a CSV already